modules/stochastic_tools/src/samplers/CSVSampler.C

```diff
@@ -0,0 +1,69 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CSVSampler.h"
+#include "DelimitedFileReader.h"
+
+registerMooseObject("StochasticToolsApp", CSVSampler);
+
+InputParameters
+CSVSampler::validParams()
+{
+  InputParameters params = Sampler::validParams();
+  params.addClassDescription("Sampler that reads samples from CSV file.");
+  params.addRequiredParam<FileName>("samples_file",
+                                    "Name of the CSV file that contains the samples matrix.");
+  params.addParam<std::vector<dof_id_type>>(
+      "column_indices",
+      "Column indices in the CSV file to be sampled from. Number of indices here will be the same "
+      "as the number of columns per matrix.");
```

Comment:
Can you explain what this is used for?

Reply:
I think this will be convenient if the user wants to use a subset of the columns in the data file. For example, they generated samples for 10 parameters, but they want to sample just 6 parameters, they don't have to go back and delete the other four columns.

Reply:
Should it use column names?

Reply:
We could use column names from the header if provided, but I thought indices would be fine for now. I can modify it to use column names too if you think that's needed.

Reply:
If you are willing I think name support would be nice. It is a bit more flexible in the long run, you can drop in any file as long as it has those names and order of the columns is not a problem.

Reply:
Makes sense. Added that support too. This should work for both indices and names regardless of their order. They can also be repeated (i.e. use the same column for multiple parameters). Also, I am relying on the errors in DelimitedFileReader if indices or names don't match up with those in the CSV file.